src/app/tabs/camera.cpp

```diff
@@ -0,0 +1,143 @@
+}
+
+QWidget* CameraTab::cam_widget()
+{
+    QWidget* widget = new QWidget(this);
+    QVBoxLayout *layout = new QVBoxLayout(widget);
+
+    QHBoxLayout *connection = new QHBoxLayout;
+    status = new QLabel("Disconnected", widget);
+    status->setFont(Theme::font_16);
+    QPushButton *disconnect = new QPushButton("disconnect");
+    disconnect->setFont(Theme::font_14);
+    disconnect->setFlat(true);
+    disconnect->setIconSize(Theme::icon_36);
+    Theme::get_instance()->add_button_icon("wifi", disconnect);
+    connection->addWidget(status);
+    connection->addStretch();
+    connection->addWidget(disconnect);
+
+    connect(disconnect, &QPushButton::clicked, this, &CameraTab::disconnect_stream);
+
+    video_widget = new QVideoWidget;
+
+    layout->addWidget(video_widget);
+    layout->addLayout(connection);
+
+    return widget;
+}
+
+
+QWidget *CameraTab::connect_widget()
+{
+    QWidget *widget = new QWidget(this);
+    QVBoxLayout *layout = new QVBoxLayout(widget);
+
+    QLabel *label = new QLabel("connect camera stream for camera view", widget);
+    label->setFont(Theme::font_16);
+    label->setAlignment(Qt::AlignCenter);
+
+    Config *config = Config::get_instance();
+    QLineEdit *stream_url_input = new QLineEdit(config->get_cam_stream_url(), widget);
+    stream_url_input->setFixedWidth( 18*30 );
+    stream_url_input->setFont(Theme::font_18);
+    stream_url_input->setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed);
+    connect(stream_url_input, &QLineEdit::editingFinished, this, [config,stream_url_input]() {config->set_cam_stream_url(stream_url_input->text());});
+
+    status2 = new QLabel("", widget);
+    status2->setFont(Theme::font_16);
+    status2->setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed);
+
+    QPushButton *button = new QPushButton("connect", widget);
+    button->setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Preferred);
+    button->setFont(Theme::font_14);
+    button->setFlat(true);
+    button->setIconSize(Theme::icon_36);
+    Theme::get_instance()->add_button_icon("wifi", button);
+    connect(button, &QPushButton::clicked, this, [this, stream_url_input](){this->connect_stream(stream_url_input->text());});
+
+    connect(this, &CameraTab::stream_connecting, this, [button, stream_url_input]() {button->setEnabled(false); stream_url_input->setEnabled(false);} );
+    connect(this, &CameraTab::stream_disconnected, this, [button, stream_url_input]() {button->setEnabled(true); stream_url_input->setEnabled(true);} );
+
+    layout->addStretch();
+    layout->addWidget(label, 1);
+    layout->addStretch();
+    layout->addWidget(stream_url_input, 0, Qt::AlignCenter);
+    layout->addWidget(status2, 0, Qt::AlignCenter);
+    layout->addStretch();
+    layout->addWidget(button, 1, Qt::AlignCenter);
+
+    return widget;
+}
+
+void CameraTab::changed_status(QMediaPlayer::MediaStatus mediaStatus)
+{
+    qInfo() << "Camera status changed to: " << mediaStatus;
+
+    switch(mediaStatus)
+    {

```

Comment:
i know there is no precedence for this, but could you have the bracket on the same line (similar to an if block)

Reply:
I've never known anyone who sometimes prefers same line and sometimes next line. Usually people believe strongly one way or the other.

Reply:
But yes, changing now.